src/FieldWarning/Assets/Units/UnitDispatcher.cs

```diff
@@ -146,8 +147,62 @@ public void SetSelected(bool selected, bool justPreviewing)
 
         public float GetHealth() => _healthComponent.Health;
         public float MaxHealth => _movementComponent.Data.maxHealth;
-        public void HandleHit(float receivedDamage) =>
-                _healthComponent.HandleHit(receivedDamage);
+        
+        // TODO: move HandelHit to ArmorComponent
+
+        /// <summary>
+        /// Calculate the total damage dealt within a successful hit, then update health, armor and ERA values accordingly
+        /// </summary>
+        /// <param name="receivedDamage"></param>
+        public void HandleHit(WeaponData.WeaponDamage receivedDamage, float? distanceToTarget, float? distanceToCentre)
+        {
+            // First we put up a Target struct to be used in the damage calculations
+            Damage.Target unitAsTarget = new Damage.Target();
+
+            // TODO: implement armor and ERA systems
+            unitAsTarget.Armor = 0.0f;
+            unitAsTarget.EraData = new Damage.Era();
+
+            unitAsTarget.Health = _healthComponent.Health;
+
+            Damage.Target finalState = unitAsTarget;
+
+            // Calculate its damage using its damage type
+            switch (receivedDamage.DamageType)
+            {
+                case DamageTypes.KE:
+                    KEDamage keDamage = new KEDamage (
+                        receivedDamage.KineticData.GetValueOrDefault(),
+                        unitAsTarget, distanceToTarget.GetValueOrDefault()
+                    );
+                    finalState = keDamage.CalculateDamage();
```

Comment:
This introduces a bug; now the weapons are being shot and not doing any damage. The targeting component needs to account for whatever minimal distance the damage system uses.